### ISSUE TITLE
tfx doesn't install cleanly on macos (10.14.6)

### ISSUE THREAD

**Issue by apsharma-roku** [tensorflow/tfx#2473]
pip install tensorflow==2.2.0 tfx==0.22.0 (from tutorials)

```
(venv) ➜  tfx-playground-dev pip install tensorflow==2.2.0 tfx==0.22.0
Collecting tensorflow==2.2.0
  Using cached tensorflow-2.2.0-cp38-cp38-macosx_10_11_x86_64.whl (175.4 MB)
Collecting tfx==0.22.0
  Using cached tfx-0.22.0-py3-none-any.whl (1.5 MB)
Collecting protobuf>=3.8.0
  Using cached protobuf-3.13.0-cp38-cp38-macosx_10_9_x86_64.whl (1.3 MB)
Collecting keras-preprocessing>=1.1.0
  Using cached Keras_Preprocessing-1.1.2-py2.py3-none-any.whl (42 kB)
Collecting termcolor>=1.1.0
  Using cached termcolor-1.1.0.tar.gz (3.9 kB)
Collecting numpy<2.0,>=1.16.0
  Using cached numpy-1.19.2-cp38-cp38-macosx_10_9_x86_64.whl (15.3 MB)
Collecting google-pasta>=0.1.8
  Using cached google_pasta-0.2.0-py3-none-any.whl (57 kB)
Collecting tensorboard<2.3.0,>=2.2.0
  Using cached tensorboard-2.2.2-py3-none-any.whl (3.0 MB)
Collecting grpcio>=1.8.6
  Using cached grpcio-1.32.0-cp38-cp38-macosx_10_9_x86_64.whl (3.3 MB)
Collecting absl-py>=0.7.0
  Using cached absl_py-0.10.0-py3-none-any.whl (127 kB)
Collecting astunparse==1.6.3
  Using cached astunparse-1.6.3-py2.py3-none-any.whl (12 kB)
Collecting h5py<2.11.0,>=2.10.0
  Using cached h5py-2.10.0-cp38-cp38-macosx_10_9_x86_64.whl (3.0 MB)
Collecting opt-einsum>=2.3.2
  Using cached opt_einsum-3.3.0-py3-none-any.whl (65 kB)
Collecting wheel>=0.26; python_version >= "3"
  Using cached wheel-0.35.1-py2.py3-none-any.whl (33 kB)
Collecting tensorflow-estimator<2.3.0,>=2.2.0
  Using cached tensorflow_estimator-2.2.0-py2.py3-none-any.whl (454 kB)
Collecting wrapt>=1.11.1
  Using cached wrapt-1.12.1.tar.gz (27 kB)
Collecting six>=1.12.0
  Using cached six-1.15.0-py2.py3-none-any.whl (10 kB)
Collecting scipy==1.4.1; python_version >= "3"
  Using cached scipy-1.4.1-cp38-cp38-macosx_10_9_x86_64.whl (28.8 MB)
Collecting gast==0.3.3
  Using cached gast-0.3.3-py2.py3-none-any.whl (9.7 kB)
Collecting google-api-python-client<2,>=1.7.8
  Using cached google_api_python_client-1.11.0-py2.py3-none-any.whl (60 kB)
ERROR: Could not find a version that satisfies the requirement ml-metadata<0.23,>=0.22 (from tfx==0.22.0) (from versions: 0.12.0.dev0, 0.13.0.dev0, 0.13.1.dev0)
ERROR: No matching distribution found for ml-metadata<0.23,>=0.22 (from tfx==0.22.0)
```

**Comment by chongkong** [comment]
Hi. This is because TFX is not yet supprting python 3.8 yet. Your error message is because ml-metadata has not released 0.22 version with python3.8. The good news is that py38 supprt is coming in the next release (0.24.0) which will come soon.

**Comment by apsharma-roku** [comment]
Thanks for the update on this.
